Scripts/Runtime/Data/Collections/DeferredNativeArray.cs

```diff
@@ -52,6 +78,7 @@ internal unsafe struct BufferInfo
             [NativeDisableUnsafePtrRestriction] public void* Buffer;
             public int Length;
             public Allocator DeferredAllocator;
+            public bool IsPending;
```

Comment:
Naming is a bit misleading. Maybe `CanBufferAllocate`, `IsClearPending`, etc...?

`IsPending` just makes it seem like an action is imminent which isn't really the case unless you actually call createDeferred again.

I wonder too if this should be compiler defined out since it's really just a safety check. Although, if serialized, that could lead to incompatible saves between debug and release mode.